File: lista-de-exercicios/exercicio-14/ConversorHoras.test.ts
import { jest } from '@jest/globals';
import { ConversorHoras } from './ConversorHoras';

const conversorHoras = new ConversorHoras(7023);

describe('Classe ConversorHoras', () => {
    test('Deve receber os segundos que sobraram', () => {
        expect(conversorHoras.receberSegundos()).toBe(3);
    })

    test('Deve receber os minutos totais', () => {
        expect(conversorHoras.converterMinutos()).toBe(117);
    })

    test('Deve receber os minutos que sobraram', () => {
        expect(conversorHoras.receberMinutos()).toBe(57);
    })

    test('Deve receber as horas', () => {
        expect(conversorHoras.converterHoras()).toBe(1);
    })

    test('Deve receber o tempo em horas, minutos e segundos', () => {
        expect(conversorHoras.getHoras()).toBe('1h57min3seg');
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio14/ConversorHorasTest.java
package org.example.exercicio14;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class ConversorHorasTest {
    ConversorHoras conversorHoras = new ConversorHoras(7023);

    @DisplayName("Deve receber os segundos que sobraram")
    @Test
    void deveReceberSegundosSobra() {
        assertEquals(3, conversorHoras.receberSegundos());
    }
    @DisplayName("Deve receber os minutos totais")
    @Test
    void deveReceberMinutosTotais() {
        assertEquals(117, conversorHoras.converterMinutos());
    }

    @DisplayName("Deve receber os minutos que sobraram")
    @Test
    void deveReceberMinutosSobra() {
        assertEquals(57, conversorHoras.receberMinutos());
    }

    @DisplayName("Deve receber as horas")
    @Test
    void deveReceberHorasTotais() {
        assertEquals(1, conversorHoras.converterHoras());
    }

    @DisplayName("Deve receber o tempo em horas, minutos e segundos")
    @Test
    void DeveRetornarTempoEmHorasMinutosSegundos() {
        assertEquals("1h57min3seg", conversorHoras.getHoras());
    }
}